src/System.Linq/tests/ToLookupTests.cs

```diff
@@ -214,5 +214,152 @@ public void NullElementSelectorExplicitComparer()
+                });
+
+            var expected = new []
+            {
+                new RoleMetadata {Role = new Role {Id = 1}, CountA = 1667, CountrB = 0 },
+                new RoleMetadata {Role = new Role {Id = 2}, CountA = 0, CountrB = 1667 },
+                new RoleMetadata {Role = new Role {Id = 3}, CountA = 0, CountrB = 1666 }
+            };
+
+            Assert.Equal(expected, result);
+        }
+
+        [Fact]
+        public void ExcludeNullGroupResultsWithList()
+        {
+            //Create test data
+            var roles = new List<Role>
+            {
+                new Role { Id = 1 },
+                new Role { Id = 2 },
+                new Role { Id = 3 },
+            };
+
+            var memberships = Enumerable.Range(0, 5000).Select(i => new Membership
+            {
+                Id = i,
+                Role = roles[i % 3],
+                CountMe = i % 3 == 0
+            });
+
+            //Run actual test
+            var result = memberships.GroupBy(
+                m => m.Role,
+                (role, mems) => new RoleMetadata
+                {
+                    Role = role,
+                    CountA = mems.Count(m => m.CountMe),
+                    CountrB = mems.Count(m => !m.CountMe)
+                }).ToList();
+
+            var expected = new []
+            {
+                new RoleMetadata {Role = new Role {Id = 1}, CountA = 1667, CountrB = 0 },
+                new RoleMetadata {Role = new Role {Id = 2}, CountA = 0, CountrB = 1667 },
+                new RoleMetadata {Role = new Role {Id = 3}, CountA = 0, CountrB = 1666 }
+            };
+
+            Assert.Equal(expected, result);
+        }
+
+        [Fact]
+        public void ExcludeNullGroupResultsWithArray()
+        {
+            //Create test data
+            var roles = new List<Role>
+            {
+                new Role { Id = 1 },
+                new Role { Id = 2 },
+                new Role { Id = 3 },
+            };
+
+            var memberships = Enumerable.Range(0, 5000).Select(i => new Membership
+            {
+                Id = i,
+                Role = roles[i % 3],
+                CountMe = i % 3 == 0
+            });
+
+            //Run actual test
+            var result = memberships.GroupBy(
+                m => m.Role,
+                (role, mems) => new RoleMetadata
+                {
+                    Role = role,
+                    CountA = mems.Count(m => m.CountMe),
+                    CountrB = mems.Count(m => !m.CountMe)
+                }).ToArray();
+
+            var expected = new []
+            {
+                new RoleMetadata {Role = new Role {Id = 1}, CountA = 1667, CountrB = 0 },
+                new RoleMetadata {Role = new Role {Id = 2}, CountA = 0, CountrB = 1667 },
+                new RoleMetadata {Role = new Role {Id = 3}, CountA = 0, CountrB = 1666 }
+            };
+
+            Assert.Equal(expected, result);
+        }
```

Comment:
These three tests appear to be almost identical, with the difference being whether the result is g, g.ToArray(), and g.ToList(), right?  If so, can we just make it a [Theory] with InlineDatas for each of the three cases?  It could just be an ```int mode``` arg that's switched on in the body to determine which to assign to result.